src/routes/learn/$slug/-components/video-header.tsx

```diff
@@ -1,46 +1,159 @@
-import { Badge } from "~/components/ui/badge";
-import { BookOpen, Clock, Lock } from "lucide-react";
-import { type Segment } from "~/db/schema";
-import { AdminControls } from "./admin-controls";
+import {
+  BookOpen,
+  Clock,
+  Edit,
+  Lock,
+  Trash2,
+  GraduationCap,
+} from 'lucide-react';
+import { Badge } from '~/components/ui/badge';
+import { Button } from '~/components/ui/button';
+import { Link } from '@tanstack/react-router';
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from '~/components/ui/alert-dialog';
+import { useLoaderData } from '@tanstack/react-router';
+import { toast } from '~/hooks/use-toast';
+import { useNavigate } from '@tanstack/react-router';
+import { useSuspenseQuery } from '@tanstack/react-query';
+import { getCompletedUserProgressForSegmentFn } from '~/fn/progress';
+import { deleteSegmentFn } from './delete-video-button';
 
-interface VideoHeaderProps {
-  currentSegment: Segment;
-  isAdmin: boolean;
-}
+export function VideoHeader() {
+  const navigate = useNavigate();
+  const { segment, isAdmin } = useLoaderData({
+    from: '/learn/$slug/_layout/',
+  });
+
+  const { data: progress } = useSuspenseQuery({
+    queryKey: ['progress', segment.id] as const,
+    queryFn: ({ queryKey }) =>
+      getCompletedUserProgressForSegmentFn({
+        data: { segmentId: queryKey[1] },
+      }),
+  });
```

Comment:
Here and in the `loader`
Can you extract the object inside into it's own function like done here - https://github.com/webdevcody/tanstack-course-platform/blob/7e591ae9fc824f8add7d13720a24f56e79e9f974/src/lib/queries/comments.ts#L4-L9

This will reduce the code duplication